course_info/tests/jasmine/controllers/DetailsControllerSpec.js

```diff
@@ -212,49 +214,337 @@ describe('Unit testing DetailsController', function () {
+            $httpBackend.flush(2);
+            dc.submitCourseDetailsForm();
+            $httpBackend.expectPATCH(courseInstanceURL, expectedPatchData).respond(500);
+            $httpBackend.flush(1);
+            expect(dc.showNewGlobalAlert).toHaveBeenCalledWith('updateFailed', '');
+            expect(dc.courseInstance['title']).not.toBe('test title');
+        });
     });
 
     describe('editableInputDirective', function() {
-        it('should substitute the right stuff for id, name, and label');
+        var compile, scope, directiveElem;
+        beforeEach(function () {
+             inject(function($compile, $rootScope){
+                compile = $compile;
+                scope = $rootScope.$new();
+              });
+              directiveElem = getCompiledElement();
+        });
+
+        function getCompiledElement(){
+            var element = angular.element('<hu-editable-input ' +
+                    'editable="dc.editable"' +
+                    'is-loading="dc.isUndefined(dc.courseInstance.term)"' +
+                    'form-value="dc.courseInstance.term"' +
+                    'model-value="testModelValue"' +
+                    'maxlength="500"' +
+                    'field="test-name"' +
+                'label="test label">' +
+                '</hu-editable-input>');
+            var compiledElement = compile(element)(scope);
+            //scope.$digest();
+            scope.$apply();
+            return compiledElement;
+        }
+
+        it('should substitute the right stuff for id, name, and label', function(){
+            dc = $controller('DetailsController', {$scope: scope});
+            $httpBackend.expectGET(courseInstanceURL).respond(200, JSON.stringify({status: "success"}));
+            $httpBackend.expectGET(peopleURL).respond(200, JSON.stringify({status: "success"}));
+            $httpBackend.flush(2);
+            var liElement = directiveElem.find('li');
+            expect(liElement.find('label').text().trim()).toBe('test label');
+            expect(liElement.find('input').prop('id')).toBe('input-course-test-name');
+            expect(liElement.find('input').prop('maxlength')).toBe(500);
+        });
+
         it('should show an input element if called with editable=true ' +
             'and the value should be equal to the form\'s copy of the ' +
-            'model data');
-        it('should show regular non-editable text if called with ' +
+            'model data', function(){
+            dc = $controller('DetailsController', {$scope: scope});
+            $httpBackend.expectGET(courseInstanceURL).respond(200, JSON.stringify({status: "success"}));
+            $httpBackend.expectGET(peopleURL).respond(200, JSON.stringify({status: "success"}));
+            $httpBackend.flush(2);
+            var liElement = directiveElem.find('li');
+            expect(liElement.find('input').prop('id')).toBe('input-course-test-name');
```

Comment:
This doesn't appear to be checking the `value` of the input, but just its `id`; also shouldn't we be checking if the `input` is visible and the `span` isn't, to meet the `it()` criteria? That may be hard to do (based on the comments starting on line 489, in the test below this), so we might have to 'tech debt' that part of the test as well.